openh264/src/encoder.rs

```diff
@@ -158,6 +158,9 @@ impl Encoder {
 
     /// Encodes a YUV source and returns the encoded bitstream.
     ///
+    /// The first source the Encoder encodes will return a keyframe ("I" frame); subsequent calls to this function will produce "difference" frames.
```

Comment:
I still find this a bit confusing. How about

The returned bitstream consists of one or more NAL units or packets. The first packets contain initialization information. Subsequent packages then contain, amongst others, keyframes ("I frames") or delta frames. 
The interval at which they are produced depends on the encoder settings.

Reply:
Well, it seems that I haven't really understood how the encoder works, then. Is what you described the real behavior? Also, I have missed which encoder settings affect the rate at which keyframes are generated.

Reply:
>  Is what you described the real behavior?

Not a H.264 expert, but what I described is pretty much any stream I've ever seen. Starts with SPS (sequence parameter set), followed by PPS (picture parameter set), sometimes some encoder specific packets, then a never-ending sequence of key frames and "delta frames". The type of "delta frame" depends on the encoder settings. 

> Also, I have missed which encoder settings affect the rate at which keyframes are generated.

I don't know the exact name in OpenH264, but it usually goes by GOP (group of picture), but depends on a few settings how exactly that happens. 

We should probably expose and document that setting.

Reply:
I'm far from being an expert myself, but my experience using this (though only once) was that only the first call to `encode` generated a keyframe. Given that those must be generated every once in a while, an important question is whether they can be generated by calls other than the first one.

Reply:
Ah sorry, I answered how the encoder works in general. Specifically encode can return more than one NAL (packet) per call (eg SPS, PPS and first key frame usually go together). Subsequent encode call would usually return delta frames based on GOP. 

That said I don't want to rule out bugs. 

What I recommend: Encode 200 frames or so and concat the output into a .h264 file. The open that file with H264 Analyzer and see what it says. If you don't see at least a few I frames there is a bug.

Reply:
I can do that. Is there a way to get the SPS and PPS separately, though? The `mp4` crate [requires them outside of the stream](https://github.com/ISSOtm/gb-packing-visualizer/blob/f9d30533794596eb626f622f68d7e31cc028537b/src/render.rs#L263-L269) (please excuse my terrible, clueless programming).

Reply:
You get all of that from this crate, and it should be simple:

In a loop, call encode, and on the returned struct call .to_vec() which returns a byte array. Just keep concatenating these arrays into a file with the .h264 extension, and you have a video. That video you can inspect, or even play in VLC.

Reply:
Agreed, it's what I have done at first, but it's not a common format so I wanted to wrap it in a MP4 file to improve compatibility. You could say that format conversion should be handled externally (e.g. `ffmpeg`, and I'll accept that.

Reply:
> You could say that format conversion should be handled externally (e.g. ffmpeg, and I'll accept that.

Yes, MP4 & friends should be handled by other crates. That said, if you just want the SPS and PPS, I've added a (very experimental) unit test `encoder_sps_pps` which shows how to best get them right now.

Reply:
Okay, then I think adding your initial suggestion instead of mine, and mentioning how to get the SPS and PPS (likely in the documentation of `EncodedBitStream`?) would do the job.